spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/test/java/com/example/FirestoreSampleApplicationNativeIntegrationTests.java

```diff
@@ -88,48 +106,45 @@ static void beforeAll() {
 
   @Test
   void testApplicationNativeDockerImage() throws ExecutionException, InterruptedException {
-    // expose firestore port from local machine to be used by our app's docker image
-    Integer firestorePort = emulator.getMappedPort(8080);
-    org.testcontainers.Testcontainers.exposeHostPorts(firestorePort);
-
-    try (MyAppNativeDockerContainer myApp = new MyAppNativeDockerContainer(emulator)) {
-      myApp.start();
-      RestTemplate restTemplate = new RestTemplateBuilder().rootUri(myApp.getBaseUrl()).build();
-      TestUserClient testUserClient = new TestUserClient(restTemplate);
-
-      testUserClient.removePhonesForUser("Alpha");
-      List<User> allUsers = testUserClient.listUsers();
-      assertThat(allUsers).isEmpty();
-
-      testUserClient.saveUser(ALPHA_USER, ALPHA_PHONE_NUMBERS);
-      verifyUserPersisted(ALPHA_USER);
-
-      testUserClient.saveUser(BETA_USER, emptyList());
-      verifyUserPersisted(BETA_USER);
-
-      testUserClient.saveUser(DELTA_USER, emptyList());
-      verifyUserPersisted(DELTA_USER);
-
-      allUsers = testUserClient.listUsers();
-      assertThat(allUsers).map(User::getName).containsExactlyInAnyOrder("Alpha", "Beta", "Delta");
-
-      List<User> users49 = testUserClient.findUsersByAge(49);
-      assertThat(users49).containsExactlyInAnyOrder(ALPHA_USER, DELTA_USER);
-      List<PhoneNumber> phoneNumbers = testUserClient.listPhoneNumbers("Alpha");
-      assertThat(phoneNumbers)
-          .map(PhoneNumber::getNumber)
-          .containsExactlyInAnyOrder("555666777", "777666555");
-
-      testUserClient.removeUserByName("Alpha");
-      phoneNumbers = testUserClient.listPhoneNumbers("Alpha");
-      assertThat(phoneNumbers)
-          .map(PhoneNumber::getNumber)
-          .containsExactlyInAnyOrder("555666777", "777666555");
-
-      testUserClient.removePhonesForUser("Alpha");
-      phoneNumbers = testUserClient.listPhoneNumbers("Alpha");
-      assertThat(phoneNumbers).isEmpty();
-    }
+    TestUserClient testUserClient = new TestUserClient(testRestTemplate.getRestTemplate());
+
+    testUserClient.removePhonesForUser("Alpha");
+    List<User> allUsers = testUserClient.listUsers();
+    assertThat(allUsers).isEmpty();
+
+    testUserClient.saveUser(ALPHA_USER, ALPHA_PHONE_NUMBERS);
+    verifyUserPersisted(ALPHA_USER);
+
+    testUserClient.saveUser(BETA_USER, emptyList());
+    verifyUserPersisted(BETA_USER);
+
+    testUserClient.saveUser(DELTA_USER, emptyList());
+    verifyUserPersisted(DELTA_USER);
+
+    User savedUser = testUserClient.saveUser(NULL_NAME_USER, emptyList());
```

Comment:
Would it be possible to just have minimal changes for the `NULL_USER` case and assertion added here in [`FirestoreSampleNativeApplicationIntegrationTests`](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/test/java/com/example/FirestoreSampleNativeApplicationIntegrationTests.java) (maybe also in [`FirestoreSampleApplicationIntegrationTests`](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample/src/test/java/com/example/FirestoreSampleApplicationIntegrationTests.java) if it’s not too much effort to do), without removing the `MyAppNativeDockerContainer` aspect of this sample test? 

Please let me know if I am missing any context for why the other changes were made here; if this ends up not being a simple fix, then feel free to leave these tests for the samples as-is, since I think validating for this behaviour in `FirestoreTemplateTests` should be sufficient.

Reply:
Thanks @emmileaf , I could not execute the original `FirestoreSampleApplicationNativeIntegrationTest` in my local machine as the test depends on a `spring-cloud-gcp-data-firestore-sample:test` image which is not accessible to me. I realized that I could simplify the test to depend instead on the codebase available with the spring-cloud-gcp-data-firestore-sample instead of an image packaged with the binary of the codebase. I also added the null check as part of it as then this feature gets integration tested as part of the change - the issue had mentioned the scenario of emulator, so that is one reason to fix this test. Let me know, I can remove this if it still feels not relevant.

Reply:
This one is our bad. `FirestoreSampleApplicationNativeIntegrationTests` has not worked in some time; we should have removed it since we'll be rewriting native support across Spring Cloud GCP.

You might notice that this test does not run with `-Dit.firestore=true` -- could you port your changes to `FirestoreSampleApplicationIntegrationTests`, so they are part of our normal CI?

Reply:
Done @elefeint, I have now reverted the changes to `FirestoreSampleApplicationNativeIntegrationTests` and instead added the test changes to `FirestoreSampleApplicationIntegrationTests`. I will let the CI verify it however as it depends on a ci project to execute the test. I am assuming Test container based tests will not run in the CI env?

Reply:
Thank you for porting the changes to `FirestoreSampleApplicationIntegrationTests`, and apologies for missing this earlier!

I ran the Firestore integration tests locally and verified that they pass –  though integration tests were not triggered as part of the presubmit checks for this PR, this test will be included as part of Firestore integration tests in our CI workflow.